pkg/autoupdate/tuf/library_manager.go

```diff
@@ -0,0 +1,341 @@
+package tuf
+
+import (
+	"context"
+	"crypto/sha512"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/fsutil"
+	"github.com/kolide/kit/version"
+	"github.com/kolide/launcher/ee/localserver"
+	"github.com/kolide/launcher/pkg/autoupdate"
+	client "github.com/theupdateframework/go-tuf/client"
+)
+
+// updateLibraryManager manages the update libraries for launcher and osquery.
+// It downloads and verifies new updates, and moves them to the appropriate
+// location in the library specified by the version associated with that update.
+// It also ensures that old updates are removed when they are no longer needed.
+type updateLibraryManager struct {
+	metadataClient  *client.Client // used to validate downloads
+	mirrorUrl       string         // dl.kolide.co
+	mirrorClient    *http.Client
+	rootDirectory   string
```

Comment:
Is this meant to be the launcher root, or the ulm root? If it's the ulm root, I'd rather call it `baseDir` (or something). And I'm not sure it should get the launcher root

Reply:
☝️ this is what I went with